keras_contrib/applications/densenet.py

```diff
@@ -258,6 +368,38 @@ def __dense_block(x, nb_layers, nb_filter, growth_rate, bottleneck=False, dropou
     return x, nb_filter
 
 
+def __transition_up_block(ip, nb_filters, type='upsampling', output_shape=None, weight_decay=1E-4):
+    ''' SubpixelConvolutional Upscaling (factor = 2)
+
+    Args:
+        ip: keras tensor
+        nb_filters: number of layers
+        type: can be 'upsampling', 'subpixel', 'deconv', or 'atrous'. Determines type of upsampling performed
+        output_shape: required if type = 'deconv'. Output shape of tensor
+        weight_decay: weight decay factor
+
+    Returns: keras tensor, after applying batch_norm, relu-conv, dropout, maxpool
+    '''
+
+    if type == 'upsampling':
+        x = UpSampling2D()(ip)
+    elif type == 'subpixel':
+        x = Convolution2D(nb_filters, 3, 3, activation="relu", border_mode='same', W_regularizer=l2(weight_decay),
+                          bias=False)(ip)
+        x = SubPixelUpscaling(r=2, channels=int(nb_filters // 4))(x)
+        x = Convolution2D(nb_filters, 3, 3, activation="relu", border_mode='same', W_regularizer=l2(weight_decay),
+                          bias=False)(x)
+    elif type == 'atrous':
+        # waiting on https://github.com/fchollet/keras/issues/4018
+        x = AtrousConvolution2D(nb_filters, 3, 3, activation="relu", W_regularizer=l2(weight_decay),
```

Comment:
```
Additionally the original DenseNet (not the tiramisu U-net version) can be updated to become FCN through atrous convolutions (non-transpose) and loading the original weights. 
``` 
How does that even work? The atrous version would perform completely different operations on the network. Loading the weights of normal densenet, even if it has the same number of layers would give completely random output.